src/bundle/Controller/UserSettingsController.php

```diff
@@ -102,6 +104,14 @@ public function updateAction(Request $request, UpdateView $view)
                     'ibexa_user_settings'
                 );
 
+                if ($form->getClickedButton() instanceof Button
+                    && $form->getClickedButton()->getName() === UserSettingUpdateType::BTN_UPDATE_AND_EDIT
```

Comment:
Nitpick: it's usually more optimal to extract `$form->getClickedButton()` instead of calling it twice, unless it's a property getter.